examples/blockchain/examples/blockchain.rs

```diff
@@ -99,21 +94,14 @@ async fn main() {
 
     let (close_member, exit) = oneshot::channel();
     tokio::spawn(async move {
-        let keybox = KeyBox {
-            count: n_members,
-            index: my_id.into(),
-        };
-        let config = aleph_bft::default_config(n_members.into(), my_id.into(), 0);
-        run_session(
-            config,
-            network,
-            data_provider,
-            finalization_provider,
-            keybox,
-            Spawner {},
-            exit,
-        )
-        .await
+        let keychain = Keychain::new(args.n_members.into(), args.my_id.into());
+        let config = aleph_bft::default_config(args.n_members.into(), args.my_id.into(), 0);
+        let units = Arc::new(Mutex::new(vec![]));
+        let unit_loader = Loader::new((*units.lock()).clone());
+        let unit_saver = Saver::new(units);
+        let local_io =
+            aleph_bft::LocalIO::new(data_provider, finalization_handler, unit_saver, unit_loader);
+        run_session(config, local_io, network, keychain, Spawner {}, exit).await
     });
 
     let mut max_block_finalized = 0;
```

Comment:
Similarly as in `dummy_honest` example, could be changed to something that would error!/panic! if finalization stream closes before we have `args.n_finalized` finalized blocks. And in future we could include it in our pipeline.

Reply:
fixed

Reply:
I would also add some message to the panic, so it prints something. Maybe just "Finalization stream finished too soon"?

Reply:
I would also add some message to the panic, so it prints something. Maybe just "Finalization stream finished too soon"?

Reply:
fixed (also in `dummy_honest.rs`)